test/extended/util/framework.go

```diff
@@ -1912,16 +1912,18 @@ func WaitForUserBeAuthorized(oc *CLI, user string, attributes *authorizationapi.
 // GetRouterPodTemplate finds the router pod template across different namespaces,
 // helping to mitigate the transition from the default namespace to an operator
 // namespace.
-func GetRouterPodTemplate(oc *CLI) (*corev1.PodTemplateSpec, string, error) {
+func GetRouterPodTemplate(oc *CLI, doesApiResourceExistFn DoesApiResourceExistFn) (*corev1.PodTemplateSpec, string, error) {
 	appsclient := oc.AdminAppsClient().AppsV1()
 	k8sappsclient := oc.AdminKubeClient().AppsV1()
 	for _, ns := range []string{"default", "openshift-ingress", "tectonic-ingress"} {
-		dc, err := appsclient.DeploymentConfigs(ns).Get(context.Background(), "router", metav1.GetOptions{})
-		if err == nil {
-			return dc.Spec.Template, ns, nil
-		}
-		if !kapierrs.IsNotFound(err) {
-			return nil, "", err
+		if doesApiResourceExistFn == nil || doesApiResourceExistFn("deploymentconfigs") {
+			dc, err := appsclient.DeploymentConfigs(ns).Get(context.Background(), "router", metav1.GetOptions{})
```

Comment:
I wonder whether any of the listed namespace is still using DC.

Reply:
removed and the PR is a lot of simpler now